File: java/parser/src/main/java/gov/lbl/parser/parser/For_indexVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Expression;
import gov.lbl.parser.domain.For_index;

public class For_indexVisitor extends modelicaBaseVisitor<For_index> {
    @Override
    public For_index visitFor_index(modelicaParser.For_indexContext ctx) {
        String identifier = ctx.IDENT().getText();
        ExpressionVisitor expressionVisitor = new ExpressionVisitor();
        Expression expression = ctx.expression() == null ? null : ctx.expression().accept(expressionVisitor);
        return new For_index(identifier, expression);
    }
}

File: jsParser/parser/For_indexVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const For_index = require('../domain/For_index')

const ExpressionVisitor = require('./ExpressionVisitor')

class For_indexVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitFor_index (ctx) {
    const identifier = ctx.IDENT() ? ctx.IDENT().getText() : ''
    let expression = null

    if (ctx.expression()) {
      const expressionVisitor = new ExpressionVisitor.ExpressionVisitor()
      expression = expressionVisitor.visitExpression(ctx.expression())
    }

    return new For_index.For_index(identifier, expression)
  }
};

For_indexVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitFor_index = this.visitFor_index
exports.For_indexVisitor = For_indexVisitor
